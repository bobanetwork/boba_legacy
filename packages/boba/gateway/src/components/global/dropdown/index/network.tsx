import React, { useState, useCallback } from 'react'
import ArrowDown from '../../../../images/icons/arrowdown.svg'
import {
  DropdownContainer,
  Header,
  IconContainer,
  Option,
  DefaultIcon,
  DropdownBody,
  Icon,
  DropdownContent,
} from '../themes/networkStyles'
interface IDropdownItem {
  value: string
  label: string
  imgSrc?: string
}

export interface IDropdownProps {
  items: IDropdownItem[]
  defaultItem: IDropdownItem
  onItemSelected?: (item: IDropdownItem) => void
}

export const Dropdown: React.FC<IDropdownProps> = ({
  items,
  defaultItem,
  onItemSelected,
}) => {
  const [selectedItem, setSelectedItem] = useState<IDropdownItem>(defaultItem)
  const [isOpen, setIsOpen] = useState<boolean>(false)

  const handleDropdown = useCallback(() => {
    setIsOpen(!isOpen)
  }, [isOpen])

  const selectItem = useCallback(
    (item: IDropdownItem) => {
      setSelectedItem(item)
      setIsOpen(false)
      onItemSelected && onItemSelected(item)
    },
    [onItemSelected]
  )

  return (
    <DropdownContainer className="dropdown">
      <Header onClick={handleDropdown}>
        <Option>
          {selectedItem.imgSrc && (
            <IconContainer>
              {selectedItem.imgSrc !== 'default' && (
                <Icon src={selectedItem.imgSrc} alt={selectedItem.label} />
              )}
              {selectedItem.imgSrc === 'default' && <DefaultIcon />}
            </IconContainer>
          )}
          {selectedItem.label}

          {isOpen ? <img src={ArrowDown} /> : <img src={ArrowDown} />}
        </Option>
      </Header>
      {isOpen && (
        <DropdownBody>
          <DropdownContent>
            {items.map((item, index) => (
              <Option
                key={index}
                onClick={() => selectItem(item)}
                style={{
                  flexDirection: 'row',
                  display: 'flex',
                  alignItems: 'center',
                  justifyContent: 'flex-start',
                }}
              >
                {item.imgSrc && (
                  <IconContainer style={{ alignContent: 'center' }}>
                    {item.imgSrc !== 'default' && (
                      <Icon
                        style={{ alignContent: 'center' }}
                        src={item.imgSrc}
                        alt={item.label}
                      />
                    )}
                    {item.imgSrc === 'default' && <DefaultIcon />}
                  </IconContainer>
                )}
                {item.label}
              </Option>
            ))}
          </DropdownContent>
        </DropdownBody>
      )}
    </DropdownContainer>
  )
}
