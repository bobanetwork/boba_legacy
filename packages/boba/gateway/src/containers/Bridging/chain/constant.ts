import { ElementType } from 'react'

import AvalancheIcon from 'components/icons/chain/L1/AvalancheIcon'
import BNBIcon from 'components/icons/chain/L1/BNBIcon'
import EthereumIcon from 'components/icons/chain/L1/EthereumIcon'

import BobaAvaxIcon from 'components/icons/chain/L2/BobaAvaxIcon'
import BobaBNBIcon from 'components/icons/chain/L2/BobaBNBIcon'
import BobaIcon from 'components/icons/chain/L2/BobaIcon'

type IconType = {
  L1: ElementType
  L2: ElementType
}

type NetworkIconsType = Record<string, IconType>

export const NETWORK_ICONS: NetworkIconsType = {
  ethereum: {
    L1: EthereumIcon,
    L2: BobaIcon,
  },
  bnb: {
    L1: BNBIcon,
    L2: BobaBNBIcon,
  },
  avax: {
    L1: AvalancheIcon,
    L2: BobaAvaxIcon,
  },
}
