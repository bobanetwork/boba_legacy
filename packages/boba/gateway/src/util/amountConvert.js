/*
Copyright 2019-present OmiseGO Pte Ltd

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

     http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License. */

// we use BigNumber here for decimal support
import BigNumber from 'bignumber.js';

export function logAmount (amount, power, truncate = 0) {

  const x = new BigNumber(amount);
  const exp = new BigNumber(10).pow(power);

  const calculated = x.div(exp);

  if(truncate > 0)
  	return calculated.toFixed(truncate);
  else
  	return calculated.toFixed();
}

/*Takes a value such as 3.92 and converts it into
a BigNumber in wei

Duplicates

ethers.utils.parseUnits( valueString , decimalsOrUnitName )   =>   BigNumber
*/

export function powAmount (amount, decimals) {

  const x = new BigNumber(amount)
  const exp = new BigNumber(10).pow(decimals)

  const calculated = x.multipliedBy(exp)
  return calculated.toFixed(0)
}

/* more clearly named version of this */
export function toWei_String(amount, decimals) {

  const x = new BigNumber(amount)
  const exp = new BigNumber(10).pow(decimals)

  const calculated = x.multipliedBy(exp)
  return calculated.toFixed(0)
}

export function amountToUsd(amount, lookupPrice, token) {
  if (token.symbol === 'ETH' && !!lookupPrice['ethereum']) {
    return amount * lookupPrice['ethereum'].usd
  } else if (token.symbol === 'BOBA' && !!lookupPrice[ 'boba-network' ]) {
    return amount * lookupPrice['boba-network'].usd
  } else if (token.symbol === 'OLO' && !!lookupPrice[ 'oolongswap' ]) {
    return amount * lookupPrice['oolongswap'].usd
  } else if (token.symbol === 'OMG' && !!lookupPrice[ 'omisego' ]) {
    return amount * lookupPrice['omisego'].usd
  } else if (token.symbol === 'USDC' && !!lookupPrice[ 'usd-coin' ]) {
    return amount * lookupPrice['usd-coin'].usd 
  } else if (!!lookupPrice[ token.symbol.toLowerCase() ]) {
    return amount * lookupPrice[token.symbol.toLowerCase()].usd
  } else {
    return 0
  }
}
