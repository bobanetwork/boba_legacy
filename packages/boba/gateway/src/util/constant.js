require('dotenv').config()
export const POLL_INTERVAL = process.env.REACT_APP_POLL_INTERVAL

export const BRIDGE_TYPE = {
  FAST_BRIDGE: "FAST_BRIDGE",
  CLASSIC_BRIDGE: "CLASSIC_BRIDGE",
  MULTI_BRIDGE: "MULTI_BRIDGE", /// FIXME: remove me
}
