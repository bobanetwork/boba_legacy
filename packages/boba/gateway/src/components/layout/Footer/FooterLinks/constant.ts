import { ROUTES_PATH } from 'util/constant'

interface LinkProps {
  label: string
  path: string
  isNav?: boolean
}

export const FOOTERLINKS: Array<LinkProps> = [
  {
    label: 'FAQs',
    path: 'https://docs.boba.network/faq',
  },
  {
    label: 'Dev Tools',
    path: ROUTES_PATH.DEV_TOOLS,
    isNav: true,
  },
  {
    label: 'Bobascope',
    path: ROUTES_PATH.BOBASCOPE,
    isNav: true,
  },
]

export const FOOTERLINKS_RIGHT: Array<LinkProps> = [
  {
    label: 'Boba Network Website',
    path: 'https://boba.network',
  },
  {
    label: 'Terms of Use',
    path: 'https://boba.network/terms-of-use/',
  },
]
