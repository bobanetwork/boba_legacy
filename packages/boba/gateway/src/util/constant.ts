import { addMonths, Now, addYear, Dayjs } from 'util/dates'

/**************
 * Env Params *
 **************/
/* eslint-disable @typescript-eslint/no-var-requires */
require('dotenv').config()

type EnvType = string | number | null | undefined

export const POLL_INTERVAL: EnvType =
  process.env.REACT_APP_POLL_INTERVAL || 20000
export const GAS_POLL_INTERVAL: EnvType =
  process.env.REACT_APP_GAS_POLL_INTERVAL || 40000
export const GA4_MEASUREMENT_ID: EnvType =
  process.env.REACT_APP_GA4_MEASUREMENT_ID || null
export const APP_ENV: EnvType = process.env.REACT_APP_ENV || 'dev'
export const SENTRY_DSN: EnvType = process.env.REACT_APP_SENTRY_DSN || null
export const INFURA_ID: EnvType = process.env.REACT_APP_INFURA_ID
export const MAX_HEALTH_BLOCK_LAG: EnvType =
  process.env.REACT_APP_MAX_HEALTH_BLOCK_LAG
export const WALLET_VERSION: EnvType = process.env.REACT_APP_WALLET_VERSION
export const APP_STATUS: EnvType = process.env.REACT_APP_STATUS || 'normal'
export const SPEED_CHECK: EnvType = process.env.REACT_APP_SPEED_CHECK
export const TARGET_CHAIN_URL: EnvType = process.env.REACT_APP_TARGET_CHAIN_URL
// VE DAO FLAG
export const DISABLE_VE_DAO: EnvType = process.env.REACT_APP_DISABLE_VE_DAO
// WalletConnect FLAG
export const DISABLE_WALLETCONNECT: EnvType =
  process.env.REACT_APP_DISABLE_WALLETCONNECT
export const BOBABEAM_STATUS: EnvType =
  process.env.REACT_APP_BOBA_BEAM_STATUS || 0
export const BOBAOPERA_STATUS: EnvType =
  process.env.REACT_APP_BOBA_OPERA_STATUS || 0

type BridgeType = {
  FAST_BRIDGE: string
  CLASSIC_BRIDGE: string
  MULTI_BRIDGE: string
  MULTI_CHAIN_BRIDGE: string
}

export const BRIDGE_TYPE: BridgeType = {
  FAST_BRIDGE: 'FAST_BRIDGE',
  CLASSIC_BRIDGE: 'CLASSIC_BRIDGE',
  MULTI_BRIDGE: 'MULTI_BRIDGE', //fix me remove me
  MULTI_CHAIN_BRIDGE: 'MULTI_CHAIN_BRIDGE',
}

type ExpiryOptionType = {
  value: string | Dayjs
  label: string
}

export const EXPIRY_OPTIONS: ExpiryOptionType[] = [
  {
    value: addMonths(Now(), 3, 'YYYY-MM-DD'),
    label: '3 Months',
  },
  {
    value: addMonths(Now(), 7, 'YYYY-MM-DD'),
    label: '6 Months',
  },
  {
    value: addYear(1, 'YYYY-MM-DD'),
    label: '1 Year',
  },
]

export const ECOSYSTEM_CATEGORY: string[] = [
  'defi',
  'gamefi',
  'nft',
  'bridge',
  'wallet',
  'tool',
  'token',
]

/*********************
 * Routes Constants **
 *********************/

type RoutesPathType = {
  BRIDGE: string
  ECOSYSTEM: string
  WALLET: string
  HISTORY: string
  EARN: string
  LOCK: string
  STAKE: string
  HELP: string
  BOBASCOPE: string
  MONSTER: string
  VOTE_DAO: string
  DAO: string
  DEV_TOOLS: string
}

export const ROUTES_PATH: RoutesPathType = {
  BRIDGE: '/bridge',
  ECOSYSTEM: '/ecosystem',
  WALLET: '/wallet',
  HISTORY: '/history',
  EARN: '/earn',
  LOCK: '/lock',
  STAKE: '/stake',
  HELP: '/help',
  BOBASCOPE: '/bobascope',
  MONSTER: '/monster',
  VOTE_DAO: '/votedao',
  DAO: '/DAO',
  DEV_TOOLS: '/devtools',
}

export const PER_PAGE: number = 8

type Network = 'ethereum' | 'bnb' | 'avax' | 'fantom' | 'moonbeam' //we move this to global network type once we define this
type Page =
  | 'Bridge'
  | 'Ecosystem'
  | 'Wallet'
  | 'History'
  | 'Earn'
  | 'Stake'
  | 'DAO'
  | 'Monster'
type PagesByNetworkType = Record<Network, Page[]>

export const PAGES_BY_NETWORK: PagesByNetworkType = {
  ethereum: [
    'Bridge',
    'Ecosystem',
    'Wallet',
    'History',
    'Earn',
    'Stake',
    'DAO',
    'Monster',
  ],
  bnb: ['Bridge', 'Wallet', 'Earn', 'History'],
  avax: ['Bridge', 'Wallet', 'Earn', 'History'],
  fantom: ['Bridge', 'Wallet', 'Earn', 'History'],
  moonbeam: ['Bridge', 'Wallet', 'Earn', 'History'],
}

enum Layer {
  L1 = 'L1',
  L2 = 'L2',
}
export const LAYER: { [key: string]: Layer } = Layer

type NetworkIconType = 'ethereum' | 'boba'

type NetworkNameType = {
  L1: string
  L2: string
}

type DefaultNetworkType = {
  NAME: NetworkNameType
  ICON: Record<Layer, NetworkIconType>
}

export const DEFAULT_NETWORK: DefaultNetworkType = {
  NAME: {
    L1: 'Ethereum',
    L2: 'Boba',
  },
  ICON: {
    L1: 'ethereum',
    L2: 'boba',
  },
}

export const MM_EXTENTION_URL: string =
  'https://chrome.google.com/webstore/detail/metamask/nkbihfbeogaeaoehlefnkodbefgpgknn?hl=en'

export const MIN_NATIVE_L1_BALANCE: number = 0.002

export const BANXA_URL: string = 'https://boba.banxa.com/?'
