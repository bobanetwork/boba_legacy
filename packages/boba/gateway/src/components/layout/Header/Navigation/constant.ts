import { ROUTES_PATH } from 'util/constant'
import { IMenuItem } from './types'

export const MENU_LIST: Array<IMenuItem> = [
  {
    label: 'Bridge',
    path: ROUTES_PATH.BRIDGE,
  },
  {
    label: 'History',
    path: ROUTES_PATH.HISTORY,
  },
  {
    label: 'Earn',
    path: ROUTES_PATH.EARN,
  },
  {
    label: 'Stake',
    path: ROUTES_PATH.STAKE,
  },
  {
    label: 'Dao',
    path: ROUTES_PATH.DAO,
  },
]
