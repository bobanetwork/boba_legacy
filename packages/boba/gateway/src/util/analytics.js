import ReactGA from "react-ga4"
import { GA4_MEASUREMENT_ID } from "./constant";

export const init = () => {
  if (window.location.hostname === 'localhost') {
    return null;
  }
  if (GA4_MEASUREMENT_ID) {
    // init if the GA4 Measurement Id is available.
    ReactGA.initialize(GA4_MEASUREMENT_ID);
  }
}

export const sendPageView = (path) => {
  ReactGA.send({
    hitType: "pageview",
    page: path,
    title: path
  });
}


export default {
  init,
  sendPageView
}
