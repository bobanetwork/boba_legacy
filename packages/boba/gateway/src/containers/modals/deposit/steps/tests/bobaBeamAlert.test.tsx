import React from 'react'
import { render } from '@testing-library/react'
import Theme from 'themes'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import BobaBeamAlert from '../bobaBeamAlert'
import { NETWORK } from 'util/network/network.util'
import * as constants from 'util/constant'

const mockStore = configureStore()

const renderComponent = (state: any) => {
  return render(
    <Provider
      store={mockStore({
        ui: {
          theme: 'dark',
        },
        ...state,
      })}
    >
      <Theme>
        <BobaBeamAlert />
      </Theme>
    </Provider>
  )
}

describe('BobaBeamAlert', () => {
  test('should match the snapshot when active network is ethereum', () => {
    Object.defineProperty(constants, 'BOBABEAM_STATUS', {
      value: 1,
      writable: true,
    })
    const { asFragment } = renderComponent({
      network: {
        activeNetwork: NETWORK.ETHEREUM,
      },
    })
    expect(asFragment()).toMatchSnapshot()
  })

  test('should match the snapshot when active network is moonbeam', () => {
    Object.defineProperty(constants, 'BOBABEAM_STATUS', {
      value: 1,
      writable: true,
    })
    const { asFragment } = renderComponent({
      network: {
        activeNetwork: NETWORK.MOONBEAM,
      },
    })
    expect(asFragment()).toMatchSnapshot()
  })

  test('should match the snapshot when active network is moonbeam and status is 0', () => {
    Object.defineProperty(constants, 'BOBABEAM_STATUS', {
      value: 0,
      writable: true,
    })
    const { asFragment } = renderComponent({
      network: {
        activeNetwork: NETWORK.MOONBEAM,
      },
    })
    expect(asFragment()).toMatchSnapshot()
  })
})
