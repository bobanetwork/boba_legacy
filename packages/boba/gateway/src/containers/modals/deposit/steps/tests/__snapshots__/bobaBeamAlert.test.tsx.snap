// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BobaBeamAlert should match the snapshot when active network is ethereum 1`] = `<DocumentFragment />`;

exports[`BobaBeamAlert should match the snapshot when active network is moonbeam 1`] = `
<DocumentFragment>
  .c0 {
  padding: 0px;
  margin: 0px;
  font-weight: 400;
  font-size: 0.9rem;
  line-height: 1.25;
}

.c1 {
  color: yellow;
  font-size: 16px;
  font-weight: 700;
  margin-bottom: 5px;
}

<p
    class="c0 c1"
  >
    For users of Bobabeam or Bobabeam applications you will need to transfer all your funds to Moonbeam mainnet before May 15th or risk permanently losing access to any assets on Bobabeam
  </p>
</DocumentFragment>
`;

exports[`BobaBeamAlert should match the snapshot when active network is moonbeam and status is 0 1`] = `<DocumentFragment />`;
